sys/shell/shell.c

```diff
@@ -227,7 +227,7 @@ static int readline(char *buf, size_t size)
         }
 
         int c = getchar();
-        if (c < 0) {
+        if (c < 0 || c == '\x03' || c == '\x04') {
```

Comment:
I don't think it's right to exit on ^C. We should cancel the current line and print a new prompt. This is consistent with how bash and friends behave. Like this: benemorius@a946c44c7650d2ed3fea55d429485d7f8bdc30bf

Reply:
Thanks for the suggestion. I have implemented it, I think it is more useful.